src/btree/bt_curnext.c

```diff
@@ -959,6 +959,17 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
          */
         if (!F_ISSET(cbt, WT_CBT_ITERATE_RETRY_PREV))
             ret = __wt_cursor_key_order_check(session, cbt, true);
+
+        if (need_walk) {
+            /*
+             * The bounds positioning code relies on the assumption that if we had to walk then we
+             * can't possibly have walked to the lower bound. We check that assumption here by
+             * comparing the lower bound with our current key or recno.
+             */
+            ret = __wt_compare_bounds(
+              session, cursor, &cbt->iface.key, cbt->recno, false, &key_out_of_bounds);
```

Comment:
In the case that the lower bound inclusive flag is set, the __wt_compare_bounds function will mark the cursor positioned at the lower_bound as still inside the bounds. I think we may have to do a direct check on the lower bound key itself or turn off inclusive flag temporarily.

Reply:
That's a good point, ideally I want to reuse the compare bounds function. I wonder if we have a flag set unset macro.

Reply:
b2f1be78d